src/meta/SessionMan.cpp

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/SessionMan.h"
+
+DECLARE_int32(heartbeat_interval_secs);
+
+namespace nebula {
+namespace meta {
+
+std::unordered_map<std::string, sessionRec> SessionMan::globalSessions_ = {};
+
+void SessionMan::addOrUpdateSession(const std::string& addr, int64_t sessionId, int64_t startTime,
+                                    int64_t updateTime) {
+    auto iter = globalSessions_.find(addr);
+    // addr not exists
+    if (iter == globalSessions_.end()) {
+        sessionRec sessRec;
+        sessRec[sessionId] = std::make_pair(startTime, updateTime);
+        globalSessions_.emplace(addr, sessRec);
+    } else {
+        auto idMap = iter->second;
+        auto idIter = idMap.find(sessionId);
+        // SessionId not exists
+        if (idIter == idMap.end()) {
+            globalSessions_[addr][sessionId] = std::make_pair(startTime, updateTime);
+        } else {
+        // Update update time
+            if (globalSessions_[addr][sessionId].first == startTime) {
+                globalSessions_[addr][sessionId].second = updateTime;
+            } else {
+                // Previously invalid data
```

Comment:
here why do not care about invalid data? is there any chance to delete the invalid data?

Reply:
Yes, invalid data will be deleted